disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+        answer.poll = poll
+
+        return answer
+
+    def _to_dict(self) -> PollCreateAnswerPayload:
+        return {"poll_media": self.media._to_dict()}
+
+    def voters(
+        self, *, limit: Optional[int] = 100, after: Optional[Snowflake] = None
+    ) -> PollAnswerIterator:
+        """Returns an :class:`AsyncIterator` representing the users that have voted for this answer.
+
+        The ``after`` parameter must represent a member and meet the :class:`abc.Snowflake` abc.
+
+        .. note::
+
+            This method works only on PollAnswer(s) objects that originate from the API and not on the ones built manually.
+
+        Parameters
+        ----------
+        limit: Optional[:class:`int`]
+            The maximum number of results to return.
+            If ``None``, retrieves every user who voted for this answer.
+            Note, however, that this would make it a slow operation.
+            Defaults to ``100``.
+        after: Optional[:class:`abc.Snowflake`]
+            For pagination, votes are sorted by member.
+
+        Raises
+        ------
+        HTTPException
+            Getting the voters for this answer failed.
+        Forbidden
+            Tried to get the voters for this answer without the required permissions.
+        ValueError
+            You tried to invoke this method on an object that didn't originate from the API.
+
+        Yields
+        ------
+        Union[:class:`User`, :class:`Member`]
+            The member (if retrievable) or the user that has voted
+            for this answer. The case where it can be a :class:`Member` is
+            in a guild message context. Sometimes it can be a :class:`User`
+            if the member has left the guild.
+        """
+        if not (self.id is not None and self.poll and self.poll.message):
+            raise ValueError(
+                "This object was manually built. To use this method, you need to use a poll object retrieved from the Discord API."
+            )
+
+        return PollAnswerIterator(self.poll.message, self.id, limit=limit, after=after)
+
+
+class Poll:
+    """Represents a poll from Discord.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    question: Union[:class:`str`, :class:`PollMedia`]
+        The question of the poll. Currently, emojis are not supported in poll questions.
+    answers: List[Union[:class:`str`, :class:`PollAnswer`]]
+        The answers for this poll, up to 10.
+    duration: :class:`datetime.timedelta`
+        The total duration of the poll, up to 32 days. Defaults to 1 day.
+        Note that this gets rounded down to the closest hour.
+    allow_multiselect: :class:`bool`
+        Whether users will be able to pick more than one answer. Defaults to ``False``.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`.
+
+    Attributes
+    ----------
+    message: Optional[:class:`Message`]
```

Comment:
Let me know if this should still be public. Either way it shouldn't be too difficult for a user to get the message from whom a Poll originates from (given the context where you can get a Poll object).

Reply:
Having a reference back to the message could be useful I imagine, so this seems fine to me. Just guild/channel references can be a bit iffy, since they need to be rebound on reconnect